src/Uno.UI/UI/Xaml/Controls/TreeView/TreeView.cs

```diff
@@ -0,0 +1,358 @@
+				return SelectedNodes.Count > 0 ? SelectedNodes[0] : null;
+			}
+			set
+			{
+				if (SelectedNodes.Count > 0)
+				{
+					SelectedNodes.Clear();
+				}
+				if (value != null)
+				{
+					SelectedNodes.Add(value);
+				}
+			}
+		}
+
+		public IList<TreeViewNode> SelectedNodes
+		{
+			get
+			{
+				return ListControl?.ListViewModel?.SelectedNodes ??
+					m_pendingSelectedNodes;
+				// we'll treat the pending selected nodes as SelectedNodes value
+				// if we don't have a list control or a view model
+			}
+		}
+
+		public object SelectedItem
+		{
+			get
+			{
+				return SelectedItems.Count > 0 ? SelectedItems[0] : null;
+			}
+			set
+			{
+				if (SelectedItems.Count > 0)
+				{
+					SelectedItems.Clear();
+				}
+				if (value != null)
+				{
+					SelectedItems.Add(value);
+				}
+			}
+		}
+
+		public IList<object> SelectedItems => ListControl?.ListViewModel?.SelectedItems;
+
+		internal void UpdateSelection(TreeViewNode node, bool isSelected)
+		{
+			var viewModel = ListControl?.ListViewModel;
+			if (viewModel != null)
+			{
+				if (isSelected != viewModel.IsNodeSelected(node))
+				{
+					var selectedNodes = viewModel.SelectedNodes;
+					if (isSelected)
+					{
+						if (SelectionMode == TreeViewSelectionMode.Single && selectedNodes.Count > 0)
+						{
+							selectedNodes.Clear();
+						}
+						selectedNodes.Append(node);
+					}
+					else
+					{
+						var index = selectedNodes.IndexOf(node);
+						if (index > -1)
+						{
+							selectedNodes.RemoveAt(index);
+						}
+					}
+				}
+			}
+		}
+
+		public void Expand(TreeViewNode node)
+		{
+			var vm = ListControl.ListViewModel;
+			vm.ExpandNode(node);
+		}
+
+		public void Collapse(TreeViewNode node)
+		{
+			var vm = ListControl.ListViewModel;
+			vm.CollapseNode(node);
+		}
+
+		public void SelectAll()
+		{
+			var vm = ListControl.ListViewModel;
+			vm.SelectAll();
+		}
+
+		private void OnItemClick(object sender, ItemClickEventArgs args)
+		{
+			var itemInvokedArgs = new TreeViewItemInvokedEventArgs(args.ClickedItem);
+			//TODO: m_itemInvokedEventSource(*this, *itemInvokedArgs);
+		}
+
+		private void OnContainerContentChanging(object sender, ContainerContentChangingEventArgs args)
+		{
+			//	m_containerContentChangedSource(sender.as< winrt::ListView > (), args);
+		}
+
+		private void OnNodeExpanding(TreeViewNode sender, EventArgs args)
+		{
+			var treeViewExpandingEventArgs = new TreeViewExpandingEventArgs(sender);
+
+			if (ListControl != null)
+			{
+				if (ContainerFromNode(sender) is TreeViewItem expandingTvi)
+				{
+					if (!expandingTvi.IsExpanded)
+					{
+						expandingTvi.IsExpanded = true;
+					}
+
+					//Update TemplateSettings properties
+					var templateSettings = expandingTvi.TreeViewItemTemplateSettings;
+					templateSettings.ExpandedGlyphVisibility = Visibility.Visible;
+					templateSettings.CollapsedGlyphVisibility = Visibility.Collapsed;
+				}
+				//TODO: m_expandingEventSource(*this, *treeViewExpandingEventArgs);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)